hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/RowDataKeyGen.java

```diff
@@ -215,12 +215,12 @@ private static String getRecordPartitionPath(
       String partField = partFields[i];
       String partValue = StringUtils.objToString(partValues[i]);
       if (partValue == null || partValue.isEmpty()) {
-        partitionPath.append(hiveStylePartitioning ? String.format(HIVE_PARTITION_TEMPLATE, partField, DEFAULT_PARTITION_PATH) : DEFAULT_PARTITION_PATH);
+        partitionPath.append(hiveStylePartitioning ? HIVE_PARTITION_TEMPLATE + partField + DEFAULT_PARTITION_PATH : DEFAULT_PARTITION_PATH);
       } else {
         if (encodePartitionPath) {
           partValue = escapePathName(partValue);
         }
-        partitionPath.append(hiveStylePartitioning ? String.format(HIVE_PARTITION_TEMPLATE, partField, partValue) : partValue);
+        partitionPath.append(hiveStylePartitioning ? HIVE_PARTITION_TEMPLATE + partField + partValue : partValue);
```

Comment:
you can concatenate a template with object values, @wombatu-kun can you review this?

Reply:
Apologies, i pushed the performance test code. I've fixed it, please take a look.

Reply:
> you can't concatenate a template with object values, @wombatu-kun can you review this?

of course, i'll do it.